firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarState.swift

```diff
@@ -21,6 +21,8 @@ struct ToolbarState: ScreenState, Equatable {
     var isNewTabFeatureEnabled: Bool
     var canShowDataClearanceAction: Bool
     var canShowNavigationHint: Bool
+    /// Stores the alternative search engine that the user has temporarily selected (otherwise use the default)
+    let alternativeSearchEngine: OpenSearchEngine?
```

Comment:
With the search engine is only being displayed in the address toolbar, I am wondering if it would make more sense to store it in the `AddressBarState`.
Also `OpenSearchEngine` is a huge object which I think we should store in Redux. Can't we store the search engine id and retrieve the engine through the `AddressToolbarContainerModel` from the id?

Reply:
Thanks for letting me know, I'll move the search engine into AddressBarState! 👍 

That makes sense. I'm going to do what we do with Tabs and TabModels and make a SearchEngineModel specifically for redux.